test/utilities/extensionUtils.ts

```diff
@@ -187,21 +202,63 @@ export async function installExtension(extension: string, extensionsDir: string)
 }
 
 export async function installExtensions(excludeExtensions: ExtensionId[] = []): Promise<void> {
-  const workbench = await utilities.getWorkbench();
   const extensionsDir = path.resolve(path.join(EnvironmentSettings.getInstance().extensionPath));
-  const extensionsVsixs = FastGlob.sync('**/*.vsix', { cwd: extensionsDir });
+  const extensionPattern =
```

Comment:
Install extensions is no longer done through the UI, but by installing extensions directly using the installed version of vscode being used by our tests.

This code will also skip the installs if the directory `extensionsDir` is pointing to a location with extensions already installed.

This function first looks at the directory extensionsDir to find installed extensions. Taking that list in ensures that the needed extensions are present. If all needed are not present, it proceeds with the next step.

The next step is checking extensionsDir to see if it has the required installers and if not fails the run. If the needed vsix files are present the code will install each, serially.